packages/firestore/src/api/cache_config.ts

```diff
@@ -0,0 +1,152 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  IndexedDbOfflineComponentProvider,
+  MemoryOfflineComponentProvider,
+  MultiTabOfflineComponentProvider,
+  OfflineComponentProvider,
+  OnlineComponentProvider
+} from '../core/component_provider';
+
+export interface MemoryLocalCache {
+  kind: 'memory';
+  _onlineComponentProvider: OnlineComponentProvider;
+  _offlineComponentProvider: MemoryOfflineComponentProvider;
+}
+
+class MemoryLocalCacheImpl implements MemoryLocalCache {
+  kind: 'memory' = 'memory';
+  _onlineComponentProvider: OnlineComponentProvider;
+  _offlineComponentProvider: MemoryOfflineComponentProvider;
+
+  constructor() {
+    this._onlineComponentProvider = new OnlineComponentProvider();
+    this._offlineComponentProvider = new MemoryOfflineComponentProvider();
+  }
+}
+
+export interface IndexedDbLocalCache {
+  kind: 'indexeddb';
+  _onlineComponentProvider: OnlineComponentProvider;
+  _offlineComponentProvider: OfflineComponentProvider;
+}
+
+class IndexedDbLocalCacheImpl implements IndexedDbLocalCache {
+  kind: 'indexeddb' = 'indexeddb';
+  _onlineComponentProvider: OnlineComponentProvider;
+  _offlineComponentProvider: OfflineComponentProvider;
+
+  constructor(settings: IndexedDbSettings | undefined) {
+    let tabManager: IndexedDbTabManager;
+    if (settings?.tabManager) {
+      settings.tabManager.initialize(settings);
+      tabManager = settings.tabManager;
+    } else {
+      tabManager = indexedDbSingleTabManager(undefined);
+      tabManager.initialize(settings);
+    }
+    this._onlineComponentProvider = tabManager._onlineComponentProvider!;
+    this._offlineComponentProvider = tabManager._offlineComponentProvider!;
+  }
+}
+
+export type FirestoreLocalCache = MemoryLocalCache | IndexedDbLocalCache;
+
+// Factory function
+export function memoryLocalCache(): MemoryLocalCache {
+  return new MemoryLocalCacheImpl();
+}
+
+export interface IndexedDbSettings {
+  cacheSizeBytes?: number;
```

Comment:
Is there any reason to offer cache size config for memory local cache?

Reply:
Looks like we already offer cacheSize config for memory persistence in the lite SDK. Consider moving that setting here.

Reply:
Once we introduce memory LRU GC, then we will have a cache size. For eager GC though, size has no effect.